Formula/g/gnome-builder.rb

```diff
@@ -0,0 +1,118 @@
+class GnomeBuilder < Formula
+  desc "Develop software for GNOME"
+  homepage "https://apps.gnome.org/Builder"
+  url "https://download.gnome.org/sources/gnome-builder/46/gnome-builder-46.2.tar.xz"
+  sha256 "0c857b89003b24787f2b1d2aae12d275a074c6684b48803b48c00276d9371963"
+  license "GPL-3.0-or-later"
+
+  head do
+    url "https://gitlab.gnome.org/GNOME/gnome-builder.git", branch: "main"
+    depends_on "gom"
+  end
+
+  depends_on "desktop-file-utils" => :build
+  depends_on "gettext" => :build
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+  depends_on "portal" => :build
+  depends_on "vala" => :build
+  depends_on "cmark"
+  depends_on "editorconfig"
+  depends_on "enchant"
+  depends_on "glib"
+  depends_on "gobject-introspection"
+  depends_on "gtk4"
+  depends_on "gtksourceview5"
+  depends_on "json-glib"
+  depends_on "jsonrpc-glib"
+  depends_on "libadwaita"
+  depends_on "libgit2-glib"
+  depends_on "libpanel"
+  depends_on "libpeas"
+  depends_on "llvm"
+  depends_on "template-glib"
+  depends_on "vte3"
+
+  resource "libdex" do
+    url "https://download.gnome.org/sources/libdex/0.7/libdex-0.7.0.tar.xz"
+    sha256 "ee789f308e8edcd7dd24553bbb543f4b8e37aee64d0589936fa22b9d1bf69f15"
+
+    # macOS ucontext fix, remove in next version
+    patch do
+      url "https://gitlab.gnome.org/GNOME/libdex/-/commit/fb8727c2dee6c9636517d5e7673976621894ae5b.diff"
+      sha256 "cd8887115774aed0780b5b9896b2a93493a53b6e914937ec92ead90cd5de3b04"
+    end
+  end
+
+  # Fixes compilation failure on Linux. See:
+  # https://gitlab.gnome.org/GNOME/gnome-builder/-/issues/2176
+  patch do
+    on_linux do
+      url "https://gitlab.gnome.org/GNOME/gnome-builder/uploads/3595cbe09fb9dd41d558a317251ca356/gdkrgba.patch"
+      sha256 "5234eb22edba53a6977cd1ed9c0de667115c65bb25ddb3ce3cd7cb43cdd7592f"
+    end
+  end
+
+  # Prevents Meson from compiling schemas / updating icon cache separately from
+  # other GTK programs. We want to compile schemas / update icon cache all
+  # together
+  patch "diff --git a/meson.build b/meson.build
+index e486d52..39afd1d 100644
+--- a/meson.build
++++ b/meson.build
+@@ -388,8 +388,8 @@ subdir('po')
+ subdir('doc')
+
+ gnome.post_install(
+-  glib_compile_schemas: true,
+-  gtk_update_icon_cache: true,
++  glib_compile_schemas: false,
++  gtk_update_icon_cache: false,
+   update_desktop_database: true,
+ )"
```

Comment:
Can this use the upstream patch for this?

Reply:
Is there an upstream patch? I hadn't seen one

Reply:
There should be, since we won't merge patches that are not submitted upstream.

Reply:
An [upstream issue](https://gitlab.gnome.org/GNOME/gnome-builder/-/issues/2229) has been created.

Reply:
The upstream issue has been resolved